File: TP01/Java/TP0102.java
// 853355 Davi Puddo

import java.util.Scanner;

public class TP0102
{
	// Codificar em cifra de cesar | chave = 3
	public static String encode (String data)
	{
		String result = "";
		if (data != null)
		{
			int n = data.length();
			int buffer = 0;
			for (int i = 0; i < n; i++)
			{
				buffer = (int)data.charAt(i) + 3;
				result = result + (char)buffer;
			}
		}
		return (result);
	}

	public static void main (String[] args)
	{
		// Definir dados
		Scanner sc = new Scanner(System.in);
		String result = null;
		String data = "";

		while (!data.equals("FIM"))
		{	
			// Ler dados
			data = sc.nextLine();

			if (data != null && !data.equals("FIM"))
			{
				// Criptografar dados
				result = encode (data);
				if (result != null)
				{
					// Mostrar dados
					System.out.println (result);
				}
			}
		}
		sc = null;
	}
}

File: TP01/C/TP0102.c
// 853355 Davi Puddo

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

// Descobrir se uma cadeia de caraceteres e um palindromo
bool pali (char* s, int n)
{
	bool result = false;
	if (s)
	{
		int i = 0;
		int y = n-1;
		int w = n;

		while (i < w && s[i] == s[y])
		{
			i++;
			y--;
		}
		if (i == w)
		{
			result = true;
		}
	}
	return (result);
}

int main (void)
{
	// Definir dados
	char* data = calloc(400, sizeof(char));
	bool result = false;
	bool stop = false;
	
	if (data != NULL)
	{
		while (stop == false)
		{
			// Ler dados
			scanf(" %400[^\n]", data);
			
			if (strcmp(data, "FIM") != 0)
			{
				// Verificar se e palindromo
				result = pali(data, strlen(data));
		
				// Mostrar dados
				if (result)
				{
					printf ("SIM\n");
				}
				else
				{
					printf ("NAO\n");
				}
			}
			else
			{
				stop = true;
			}
		}
		free(data);
	}
	return (0);
}
